addons/binding/org.openhab.binding.plugwise/src/main/java/org/openhab/binding/plugwise/internal/PlugwiseCommunicationHandler.java

```diff
@@ -0,0 +1,591 @@
+                    && id.getName().equals(configuration.getSerialPort())) {
+                logger.debug("Serial port '{}' has been found", configuration.getSerialPort());
+                portId = id;
+                break;
+            }
+        }
+
+        if (portId == null) {
+            StringBuilder sb = new StringBuilder();
+            portList = CommPortIdentifier.getPortIdentifiers();
+            while (portList.hasMoreElements()) {
+                CommPortIdentifier id = (CommPortIdentifier) portList.nextElement();
+                if (id.getPortType() == CommPortIdentifier.PORT_SERIAL) {
+                    sb.append(String.format("%s%n", id.getName()));
+                }
+            }
+            throw new PlugwiseInitializationException(
+                    String.format("Serial port '%s' could not be found. Available ports are:%n%s",
+                            configuration.getSerialPort(), sb));
+        }
+
+        try {
+            serialPort = portId.open("openHAB", 2000);
+            serialPort.addEventListener(this);
+            serialPort.notifyOnDataAvailable(true);
+            serialPort.setSerialPortParams(115200, SerialPort.DATABITS_8, SerialPort.STOPBITS_1,
+                    SerialPort.PARITY_NONE);
+            outputChannel = Channels.newChannel(serialPort.getOutputStream());
+        } catch (IOException e) {
+            throw new PlugwiseInitializationException("Failed to get serial port output stream", e);
+        } catch (PortInUseException e) {
+            throw new PlugwiseInitializationException("Serial port already in use", e);
+        } catch (TooManyListenersException e) {
+            throw new PlugwiseInitializationException("Failed to add serial port listener", e);
+        } catch (UnsupportedCommOperationException e) {
+            throw new PlugwiseInitializationException("Failed to set serial port parameters", e);
+        }
+    }
+
+    public boolean isExistingMessageListener(PlugwiseMessageListener listener) {
+        for (FilteredMessageListener filteredListener : filteredListeners) {
+            if (filteredListener.listener.equals(listener)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public boolean isExistingMessageListener(PlugwiseMessageListener listener, MACAddress macAddress) {
+        for (FilteredMessageListener filteredListener : filteredListeners) {
+            if (filteredListener.listener.equals(listener) && macAddress.equals(filteredListener.macAddress)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private void notifyListeners(Message message) {
+        for (FilteredMessageListener filteredListener : filteredListeners) {
+            if (filteredListener.matches(message)) {
+                try {
+                    filteredListener.listener.handleReponseMessage(message);
+                } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception